Updated example/README (#183)
1Password/onepassword-sdk-go: example/README.md

Author: @libutcher Can you please review the README? The ones in the Python and JS repo are exactly the same besides the instructions on how to execute the code. Any changes here I'll put to the others as well.

# Examples
This folder contains a code snippet demonstrating how to use the 1Password Go SDK for performing various operations on 1Password vaults and items. Specifically, the example showcases how to:

- Authenticate with the 1Password API using a service account token.
- List available vaults and items within those vaults.
- Retrieve a specific secret and resolve a one-time password (TOTP).
- Create a new item in a vault with multiple fields and tags.
- Update an existing item by modifying its fields and adding a new website.
- Generate different types of passwords (PIN, memorable, and random).
- Share an item with valid recipients and create a shareable link.
- Archive or delete items from the vault.
- Create and manage SSH key items.

> - Create and manage document items, including replacing and reading documents.

Reviewer: We should capitalize Document here since it's an item category. 

```suggestion
- Create and manage Document items, including replacing and reading documents.
```

Also, should we separate out the read functionality, since it applies to both document items and file attachments? We could add a new bullet like:

```suggestion
- Read files stored in file fields and Document items.
```

> - Create and manage file field items by attaching and deleting files.

Reviewer: File fields are something that can be added to an item of any type, like a Login item or Database item, not an item type in themselves. So, we should say something like:

```suggestion
- Create and manage items that contain file fields, by attaching and deleting files.
```

## Prerequisites

1. Clone the repository and follow the steps to [get started](https://github.com/1Password/onepassword-sdk-go/blob/main/README.md#get-started).

> 2. Ensure that you have a valid service account token by exporting it as an environment variable:

Reviewer: Small style nit: we try to avoid "ensure" in favor of "make sure". Also, I'm wondering if this is quite accurate, since exporting a service account token doesn't in and of itself return anything about the token's validity. I think we could say something like one or the other of these:

```suggestion
2. Make sure you have a valid service account token, then export it as an environment variable:
```

```suggestion
2. Export a valid service account token as an environment variable:
```

    ```bash
    export OP_SERVICE_ACCOUNT_TOKEN="<your token>"
    ```
3. Set a vault ID that your service account has Read, Write, and Share access to as the `OP_VAULT_ID` environment variable:
    ```bash
    export OP_VAULT_ID="<your vault uuid>"
    ```

> ## How to Run

Reviewer: ```suggestion
## How to run
```

We typically sentence-case headers

To run the example file, navigate to the project root directory and run: 
```bash
go run example/main.go
```

> ## Terminal Output

Reviewer: ```suggestion
## Terminal output
```
